### PR TITLE
Remove unknown value from gender field

### LINKED ISSUE #461: [Bug]: Empty dropdown option for gender list
### Expected behavior

No empty space is displayed

Notes: it's also good to align it to capital letters (Female, Male, Others)

### Actual behavior

Empty dropdown option for gender is shown

### Steps to reproduce

1. Login with admin user
2. Navigate to `Customer Management Framework > Customers`
3. Click New Customer button
4. Click on Gender dropdown menu

empty dropdown

### PR DESCRIPTION
**Pull request by aryaantony92**
## Changes in this pull request  
Resolves https://github.com/pimcore/customer-data-framework/issues/461

### <samp>🤖 Generated by Copilot at 2324aaf</samp>

Removed the 'unknown' option from the gender field to fix a saving error. 

### <samp>🤖 Generated by Copilot at 2324aaf</samp>

> _`unknown` option_
> _buggy and unnecessary_
> _gone with autumn leaves_

### <samp>🤖 Generated by Copilot at 2324aaf</samp>

*  Removed the 'unknown' option from the gender field ([link](https://github.com/pimcore/pimcore/pull/15286/files?diff=unified&w=0#diff-2d884603802f949efc320e428d574156668273617a5d9f1df563312d4e5291ceL37-R37)) in `models/DataObject/ClassDefinition/Data/Gender.php`.

### CODE REVIEW

**Comment by NiklasBr** [comment]
Why not allow empty values?

 > On 15 June 2018, the [Austrian Constitutional Court](https://en.wikipedia.org/wiki/Austrian_Constitutional_Court) reached a decision, published in a news release on 29 June, that Article 8 of the European Convention on Human Rights guarantees recognition of gender identity beyond the binary male or female, and that people with a variation in gender development other than male or female _must be allowed to leave a gender entry empty_ and must be allowed to have a positive other entry implemented.

**Comment by aryaantony92** [comment]
> Contributor
It's about removing the highlighted empty option as shown in the image below. The option "(Empty)" would be still there if it's not a mandatory field
image

File: models/DataObject/ClassDefinition/Data/Gender.php [hunk 1]
```diff
@@ -34,8 +34,7 @@ public function configureOptions()
         $options = [
             ['key' => 'male', 'value' => 'male'],
             ['key' => 'female', 'value' => 'female'],
-            ['key' => 'other', 'value' => 'other'],
-            ['key' => '', 'value' => 'unknown'],
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Gender.php:38]: I think we should provide a migration for `unknown` value in the db. I'm not sure though if it should be converted to `null` or `other`.

**Comment by dvesh3** [comment]
@aryaantony92 why not fix it on 10.5?

**Comment by aryaantony92** [comment]
> @aryaantony92 why not fix it on 10.5?

Because there was no empty dropdown issue with 10.5

**Comment by dvesh3** [comment]
the option is there on 10.5 https://github.com/pimcore/pimcore/blob/10.5/models/DataObject/ClassDefinition/Data/Gender.php#L38

**Comment by dvesh3** [comment]
Suggestions from internal discussion: add key (with translation) for "unknown" value and don't allow this field to be marked as mandatory since the empty option should be there always.

**Comment by aryaantony92** [comment]
Closed in favour of https://github.com/pimcore/pimcore/pull/15385

**Pull request closed without merging**